exporters/otlp/internal/retry/retry.go

```diff
@@ -114,12 +114,12 @@ func (c Config) RequestFunc(evaluate EvaluateFunc) RequestFunc {
 			} else {
 				elapsed := b.GetElapsedTime()
 				if b.MaxElapsedTime != 0 && elapsed+throttle > b.MaxElapsedTime {
-					return fmt.Errorf("max retry time would elapse: %w", err)
+					return fmt.Errorf("max retry time would elapse: %w", fnErr)
 				}
 				delay = throttle
 			}
 
-			if err := waitFunc(ctx, delay); err != nil {
+			if err := waitFunc(ctx, fnErr, delay); err != nil {
```

Comment:
Keeping the error formatting in this function alone would ensure that any error returned from `waitFunc` will include the retryable error. That way when another error is returned from `wait` it won't need to duplicate the wrapping.

```suggestion
			if err := waitFunc(ctx, delay); err != nil {
				err = fmt.Errorf("%w: %s", err, fnErr)
```